internal/security/secret.go

```diff
@@ -32,34 +33,40 @@ import (
 
 // SecretProvider cache storage for the secrets
 type SecretProvider struct {
-	secretClient  pkg.SecretClient
+	SecretClient  pkg.SecretClient
```

Comment:
Why exposing this. SecretProvider should be abstracting it, so user should not need to access it directly.

Reply:
I exposed it so I could add tests to another package.  The webserver package is needs a mocked secretClient.  I don't like this approach and agree with you.  But I can't think of another way of attaching up the mocked secretClient

Reply:
arh, now we are missing the protected field from C++ LOL